gtda/homology/simplicial.py

```diff
@@ -1026,3 +1026,239 @@ def plot(Xt, sample=0, homology_dimensions=None, plotly_params=None):
+        Maximum value of the Vietoris–Rips filtration parameter. Points whose
+        distance is greater than this value will never be connected by an
+        edge, and topological features at scales larger than this value will
+        not be detected.
+
+    infinity_values : float or None, default: ``None``
+        Which death value to assign to features which are still alive at
+        filtration value `max_edge_length`. ``None`` means that this
+        death value is declared to be equal to `max_edge_length`.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    infinity_values_ : float
+        Effective death value to assign to features which are still alive at
+        filtration value `max_edge_length`.
+
+    See also
+    --------
+    VietorisRipsPersistence, FlagserPersistence, SparseRipsPersistence, \
+    EuclideanCechPersistence, ConsistentRescaling, ConsecutiveRescaling
+
+    Notes
+    -----
+    `Ripser <https://github.com/Ripser/ripser>`_ is used as a C++ backend
+    for computing Vietoris–Rips persistent homology. Python bindings were
+    modified for performance from the `ripser.py
+    <https://github.com/scikit-tda/ripser.py>`_ package.
+
+    Persistence diagrams produced by this class must be interpreted with
+    care due to the presence of padding triples which carry no information.
+    See :meth:`transform` for additional information.
+
+    References
+    ----------
+    [1] U. Bauer, "Ripser: efficient computation of Vietoris–Rips persistence \
+        barcodes", 2019; `arXiv:1908.02518 \
+        <https://arxiv.org/abs/1908.02518>`_.
+
+    """
+
+    _hyperparameters = {
+        'homology_dimensions': {
+            'type': (list, tuple),
+            'of': {'type': int, 'in': Interval(0, np.inf, closed='left')}
+            },
+        'coeff': {'type': int, 'in': Interval(2, np.inf, closed='left')},
+        'max_edge_length': {'type': Real},
+        'infinity_values': {'type': (Real, type(None))}
+        }
+
+    def __init__(self, homology_dimensions=(0, 1), coeff=2,
+                 max_edge_length=np.inf, infinity_values=None, n_jobs=None):
+        self.homology_dimensions = homology_dimensions
+        self.coeff = coeff
+        self.max_edge_length = max_edge_length
+        self.infinity_values = infinity_values
+        self.n_jobs = n_jobs
+
+    def _weak_alpha_diagram(self, X):
+        N = len(X)
+        d = Delaunay(X)
+        indptr, indices = d.vertex_neighbor_vertices
+
+        row = np.zeros_like(indices)
+        row[indptr[1:-1]] = 1
+        np.cumsum(row, out=row)
+
+        mask = indices > row
+        row, col = row[mask], indices[mask]
+        dists = np.linalg.norm(X[row] - X[col], axis=1)
+        dm = csr_matrix((dists, (row, col)), shape=(N, N))
```

Comment:
The conversion to CSR is not ideal. We would rather use COO because `ripser` converts all sparse formats to COO anyway. But there is a bug with our version of C++ ripser (now fixed upstream) which means we would get incorrect results if this function returned a COO matrix directly.

Reply:
Changed to COO following the patch in #465.